apps/web/src/components/Messaging.tsx

```diff
@@ -0,0 +1,204 @@
+  message: MessagePayload;
+  summary: string;
+}
+
+const parseComponentId = (componentId: string): ComponentId | null => {
+  if (!componentId) {
+    return null;
+  }
+
+  const [path, id, , ...ancestors] = componentId.split('##');
+  const [author, name] = path.split('/');
+  return {
+    author,
+    name,
+    path,
+    id,
+    parent: parseComponentId(ancestors.join('##')),
+  };
+};
+
+const formatProps = (props: any, isRoot = false): any => {
+  if (!props || typeof props === 'number') {
+    return props;
+  }
+
+  if (Array.isArray(props)) {
+    return `[${props.map((p) => formatProps(p)).join(', ')}]`;
+  }
+
+  if (typeof props === 'object') {
+    const formatted = Object.entries(props)
+      .map(([k, v]) => `${k}=${formatProps(v)}`)
+      .join(', ');
+    if (isRoot) {
+      return formatted;
+    }
+
+    return `{ ${formatted} }`;
+  }
+
+  return `"${props.toString()}"`;
+};
+
+const formatComponentId = (componentId: ComponentId | null) => {
+  if (!componentId) {
+    return '';
+  }
+
+  if (!componentId.id) {
+    return componentId.name;
+  }
+
+  return `${componentId.name.split('.').slice(1).join('.')}#${componentId.id}`;
+};
+
+const buildMessageSummary = (params: BWEMessage): ComponentMessage | null => {
+  const { toComponent, fromComponent, message } = params;
+  const isFromComponent = fromComponent !== undefined;
+  switch (message.type) {
+    case 'component.render': {
+      const { type, props } = message.node;
+      const formattedChildren = message.childComponents?.length
+        ? `with children ${message.childComponents
+            .map(({ componentId }) =>
+              formatComponentId(parseComponentId(componentId))
+            )
+            .join(', ')}`
+        : '';
+
+      return {
+        message,
+        isFromComponent,
+        name: 'render',
+        componentId: parseComponentId(message.componentId)!,
+        summary: `rendered <${type} ${formatProps(props, true).slice(
+          0,
+          64
+        )}...> ${formattedChildren}`,
+      };
+    }
+    case 'component.callbackInvocation': {
+      const targetComponent = formatComponentId(
+        parseComponentId(message.targetId || '')
+      );
+      const { requestId, method, args } = message;
+      return {
+        message,
+        isFromComponent,
+        name: 'invoke',
+        componentId: parseComponentId(message.originator)!,
+        summary: `[${requestId.split('-')[0]}] called ${targetComponent}.${
+          method.split('::')[0]
+        }(${args})${!isFromComponent ? ' for' : ''}`,
+      };
+    }
+    case 'component.callbackResponse': {
+      const { requestId, result } = message;
+      return {
+        message,
+        isFromComponent,
+        name: 'return',
+        componentId: parseComponentId(
+          isFromComponent ? message.containerId : message.targetId
+        )!,
+        summary: `[${requestId.split('-')[0]}] returned ${result} ${
+          !isFromComponent ? 'to' : ''
```

Comment:
I'm not sure if this stands against the convention, but might be a bit easier to read?
```suggestion
          isFromComponent ? '' : 'to'
```